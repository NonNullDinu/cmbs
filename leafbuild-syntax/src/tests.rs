fn parses_without_errors(s: &str) {
    assert_eq!(crate::parser::parse(s).errors, vec![]);
}

// Those are only *parsing* tests, so as long as they make sense
// *syntactically*, they should parse.
// Even if semantically they are void of meaning.
#[test]
fn unit_to_unit_assignment() {
    parses_without_errors("() = ()\n")
}

#[test]
fn line_trailing_space() {
    parses_without_errors("() = () \n")
}

#[test]
fn if_condition() {
    parses_without_errors("if (()) {} else {}")
}

#[test]
fn if_else_if_condition() {
    parses_without_errors("if () {} else if () {} else {}")
}

#[test]
fn if_else_if_else_condition_stretched() {
    parses_without_errors(
        r#"
    if ()

    {



    }

    else if () {


    }

    else {


    }
    "#,
    )
}

#[test]
fn if_condition_in_expr() {
    parses_without_errors("() = if () {} else {} \n")
}

#[test]
fn if_condition_in_expr_stretched() {
    parses_without_errors(
        r#"
    () = if ()
    {

    }

    else
    {


    }
    "#,
    )
}

#[test]
fn declaration() {
    parses_without_errors("let x = ()\n")
}

#[test]
fn foreach_basic() {
    parses_without_errors("foreach () in () {}")
}
