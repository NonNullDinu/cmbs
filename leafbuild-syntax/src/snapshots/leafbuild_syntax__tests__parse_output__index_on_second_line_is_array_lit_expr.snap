---
source: leafbuild-syntax/src/tests.rs
expression: "\n        x = a\n        [1]\n        "

---
- ROOT 0..35
  - "\n" NEWLINE 0..1
  - "        " WHITESPACE 1..9
  - Assignment 9..15
    - Expr 9..10
      - PrimaryExpr 9..10
        - "x" ID 9..10
    - " " WHITESPACE 10..11
    - "=" EQ 11..12
    - " " WHITESPACE 12..13
    - Expr 13..14
      - PrimaryExpr 13..14
        - "a" ID 13..14
    - "\n" NEWLINE 14..15
  - "        " WHITESPACE 15..23
  - Expr 23..26
    - PrimaryExpr 23..26
      - ArrayLitExpr 23..26
        - "[" L_BRACKET 23..24
        - Expr 24..25
          - PrimaryExpr 24..25
            - "1" NUMBER 24..25
        - "]" R_BRACKET 25..26
  - "\n" NEWLINE 26..27
  - "        " WHITESPACE 27..35

